x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java

```diff
@@ -497,7 +497,7 @@ public void resolveAsSeparateMappings(String indexWildcard, String javaRegex, bo
                     if (unmappedIndices.isEmpty() == true) {
                         concreteIndices = asList(capIndices);
                     } else {
-                        concreteIndices = new ArrayList<>(capIndices.length - unmappedIndices.size() + 1);
+                        concreteIndices = new ArrayList<>(capIndices.length);
```

Comment:
Wouldn't it make sense to have something like `Math.max(1, capIndices.length - unmappedIndices.size())`

Reply:
`capIndices` is the list of indices in which that specific field can be found, while `unmappedIndices` is the list of indices where this specific field cannot be found. And these two come from fieldcaps API. Imo, capIndices and unmappedIndices should have exclusive content and there shouldn't be any relationship between the size of one and the other.

Reply:
OK. Got it, thx!